File: src/client/java/fr/hardel/asset_editor/client/javafx/lib/action/EditorActionResult.java
package fr.hardel.asset_editor.client.javafx.lib.action;

public record EditorActionResult(Status status, String message) {

    public enum Status {
        APPLIED,
        PACK_REQUIRED,
        REJECTED,
        ERROR
    }

    public static EditorActionResult applied() {
        return new EditorActionResult(Status.APPLIED, null);
    }

    public static EditorActionResult packRequired() {
        return new EditorActionResult(Status.PACK_REQUIRED, "error:pack_required");
    }

    public static EditorActionResult rejected(String reason) {
        return new EditorActionResult(Status.REJECTED, reason);
    }

    public static EditorActionResult error(String message) {
        return new EditorActionResult(Status.ERROR, message);
    }

    public boolean isApplied() {
        return status == Status.APPLIED;
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/action/EditorActionResult.kt
package fr.hardel.asset_editor.client.compose.lib.action

data class EditorActionResult(val status: Status, val message: String?) {

    enum class Status {
        APPLIED,
        PACK_REQUIRED,
        REJECTED,
        ERROR
    }

    companion object {
        @JvmStatic
        fun applied(): EditorActionResult = EditorActionResult(Status.APPLIED, null)

        @JvmStatic
        fun packRequired(): EditorActionResult = EditorActionResult(Status.PACK_REQUIRED, "error:pack_required")

        @JvmStatic
        fun rejected(reason: String): EditorActionResult = EditorActionResult(Status.REJECTED, reason)

        @JvmStatic
        fun error(message: String): EditorActionResult = EditorActionResult(Status.ERROR, message)
    }
}
